project/src/main/webapp/usermap.js

```diff
@@ -117,6 +117,73 @@ function radiusChange(sel) {
 	displaySearch(searchQuery, currentRadius, numberOfPlaces);
 }
 
+var map;
+var autocomplete;
+var photoURLS; 
+var image;
+
+function search() {
+    // autocompleted suggestions of user searches
+    var input = document.getElementById("pac-input");
+    var options = {
+        types: ['establishment']
+    };
+    autocomplete = new google.maps.places.Autocomplete(input, options);
+    autocomplete.setFields(["place_id", "geometry"]);
+}
+
+function getCatalog(id) {
+    //returns photos associated with user search
+    autocomplete.addListener("place_changed", function() {
+        const place = autocomplete.getPlace();
+        if (!place.geometry) {
+            return;
+        }
+
+        const request = {
+            placeId: place.place_id,
+            fields: ["photos"]
+        };
+
+        service = new google.maps.places.PlacesService(map);
+        service.getDetails(request, function callback(results, status) { // this request will return an array of photos as a result
+            if (status == google.maps.places.PlacesServiceStatus.OK) {
+                photos = results.photos;
+                if (!photos) {
+                    return;
+                }
+
+                photoURLS = [];
+                console.log(photoURLS);
+                for (var i = 0; i < photos.length; i++){
+                    var url = photos[i].getUrl({maxWidth: 500, maxHeight: 500})
+                    photoURLS.push(url);
+                    console.log("added new url: ", url);
+                }
+                image = document.getElementById("image");
+                image.setAttribute("src", photoURLS[0]);
+            }
+        });
+    });
+}
+
+var i = 0;
+
+function next() {
+    i += 1;
+    if (i < photoURLS.length) {
+        image = document.getElementById("image");
+        image.setAttribute("src", photoURLS[i]);
+    }
+}
+
+function previous() {
+    i -= 1;
```

Comment:
This is an essential duplicate of what you have in usermaps.js. Can we try to dedup these somehow?

Reply:
confused about this comment

Reply:
Right now you have this global function `previous` and `next` in usermap.js. you also have the same funcitons in catalog.js.

maybe you can make a Pagination class that encapsulates these funcitons?

class Pagination {
  next()
  previous()
}

Reply:
okay, didnt meant to have does functions in 2 different files only in the catalog.js